src/client.rs

```diff
@@ -409,11 +410,39 @@ impl Client {
         let res = self.get_retry::<Row>(sql).await?;
 
         let mut next = res.next_uri;
+        let mut last_next = next.clone();
+
         while let Some(url) = &next {
             let res = self.get_next_retry::<Row>(url).await?;
-            next = res.next_uri;
+
+            let next_uri = res.next_uri;
+
+            // If next_uri is not None, update last_next
+            if next_uri.is_some() {
+                last_next = next_uri.clone();
+            }
+            next = next_uri;
+        }
+
+        let url = last_next.ok_or_else(|| {
+            Error::InternalError("No next URI available for execution result".to_string())
+        })?;
+
+        let result = self.try_get_retry_result(&url).await?;
+
+        if let Some(error) = result.error {
+            return Err(error.into());
         }
-        Ok(ExecuteResult { _m: () })
+
+        Ok(ExecuteResult { output_uri: None })
```

Comment:
Should it really always return this? 
In this case why not just have a `Ok(())`?